### PR TITLE
Add "touch" operation

### PR DESCRIPTION
**Pull request by cottsay**
This operation is intended to function similar to the "touch" command and will essentially bypass the optimizations which avoid modifying repository metadata when there are no effective changes to it.

It is particularly useful for using createrepo-agent to generate initial metadata for an empty repository without seeding it with any packages.

The actual effect of this change happens in `repo_cache.c`. All the rest of the changes are plumbing and tests.

### CODE REVIEW

File: src/createrepo-cache/repo_cache.c [hunk 1]
```diff
@@ -1996,6 +1996,26 @@ cra_sign_repomd(
   return 0;
 }
 
+int
+cra_cache_touch(cra_Cache * cache, const char * arch_name)
+{
+  cra_ArchCache * arch = NULL;
+
+  if (!arch_name) {
+    cache->source_repo->flags |= CRA_REPO_DIRTY;
```
Comment (j-rivero) [inline src/createrepo-cache/repo_cache.c:2005]: The call without arguments seems to perform a cache clear on the source_repo only. Is this intentional? If it is, would be good to document it somehow in the help. Also good if cover the no args in the tests.

**Comment by j-rivero** [comment]
Generally looks fine to my eyes. Some questions and probably considering add more tests for corner cases: no-valid-architecture argument, no-argument, single architecture passed.

> It is particularly useful for using createrepo-agent to generate initial metadata for an empty repository without seeding it with any packages.

Is this something you plan to add to the code automatically when creating the initial metadata or must be called by the user in a different command?

**Comment by j-rivero** [comment]
Other minor problem. If I run mkdir build, cd build, cmake .., make tests there is a problem on smake.touch/bar:

```
  ⎿  Error: Exit code 1                                                                                                                                                                       
     Note: Google Test filter = smoke/smoke.touch/bare                                                                                                                                        
     [==========] Running 1 test from 1 test suite.                                                                                                                                           
     [----------] Global test environment set-up.                                                                                                                                             
     [----------] 1 test from smoke/smoke                                                                                                                                                     
     [ RUN      ] smoke/smoke.touch/bare                                                                                                                                                      
     /home/jrivero/code/infra/createrepo-agent/test/integration_utils.hpp:46: Failure                                                                                                         
     Value of: fs::is_directory(fixture)                                                                                                                                                      
       Actual: false                                                                                                                                                                          
     Expected: true 
```

Seems like tests must be called from build/tests or running ctest directly?

**Comment by cottsay** [comment]
Thanks for taking a look.

> considering add more tests for corner cases: no-valid-architecture argument, no-argument, single architecture passed.

Good call out, I'll add them.

> Seems like tests must be called from build/tests or running ctest directly?

I can't reproduce this. It seems to pass for me when I run `mkdir build && cd build && cmake .. && make -j5 && make test`.

The `fixture` directory should be populated as part of the CMake configure step.

**Comment by cottsay** [comment]
> > Seems like tests must be called from build/tests or running ctest directly?
> 
> I can't reproduce this. It seems to pass for me when I run `mkdir build && cd build && cmake .. && make -j5 && make test`.
> 
> The `fixture` directory should be populated as part of the CMake configure step.

I managed to hit this! It's only the `bare` test that's failing. For some reason, the empty(-ish) `bare` fixture directory doesn't get copied into the staging directory during the configure phase.

Maybe I'll add an explicit CTest fixture that copies the fixtures into the testing directory fresh as part of each invocation.

**Comment by j-rivero** [comment]
> I managed to hit this! It's only the `bare` test that's failing. For some reason, the empty(-ish) `bare` fixture directory doesn't get copied into the staging directory during the configure phase.

great, thanks !

I think that I found a segfault happening in the tests. I've asked my debugging agent to investigate the gdb of it and the explanation that it produced sounds reasonable to me although I'm not familiar with the code base to fully verified the claims. Black-box testing the fix worked.

```
for i in $(seq 1 50); do ./test_smoke --gtest_filter=smoke/smoke.touch/bare ; done
...
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from smoke/smoke
[ RUN      ] smoke/smoke.touch/bare
CSegmentation fault (core dumped)
```

## The problem

`smoke/smoke.touch/bare` crashes non-deterministically inside libxml2 (`xmlDocDumpFormatMemoryEnc`) when called from worker threads in `cra_repomd_flush_worker` at `src/createrepo-cache/repo_cache.c:1559`.

Stack trace from the crash:

```
Thread 23 "pool" received signal SIGSEGV:
#0  libxml2.so.2 (inside xmlDocDumpFormatMemoryEnc)
#1  xmlDocDumpFormatMemoryEnc
#2  cr_xml_dump_repomd  (libcreaterepo_c)
#3  cra_xml_write_repomd
#4  cra_repomd_flush_worker
```

[libxml2's documentation](https://dev.w3.org/XInclude-Test-Suite/libxml2-2.4.24/libxml2-2.4.24/doc/html/libxml-parser.html#XMLINITPARSER) **`xmlInitParser()` must be called from the main thread before any other threads are created**. 

Why only `bare` reproduces it:

- `empty` / `populated` fixtures already have a `repomd.xml` on disk, so `cra_repo_cache_load()` → `cr_xml_parse_repomd()` runs on the main thread during realize and indirectly initializes libxml2 globals before any workers spawn.
- `bare` has no repodata, so `cra_repo_cache_realize` (`repo_cache.c:909`) short-circuits and never touches libxml2 on the main thread.
- `touch aarch64 x86_64` then dirties 5 repos (SRPMS + arch + debug × 2). `cra_cache_flush` spawns `g_get_num_processors()` workers that all race into libxml2's first-time init simultaneously → SIGSEGV.

Why it's non-deterministic: the race only fires when two workers hit the lazy init window at the same time. Single-core scheduling, CPU caches, or cold cache for libxml2's encoding tables all affect the outcome.

Why the Python test `test_touch` didn't catch it: the Python extension is loaded into an interpreter that has already pulled libxml2 state through other startup paths, papering over the race.

## A possible fix

Added a call to `cr_xml_dump_init()` (createrepo_c's thin wrapper around `xmlInitParser()`, specifically provided for this purpose) at the top of `cra_cache_new()` in `src/createrepo-cache/repo_cache.c:422`. It runs on the main thread before any flush worker pool is created, is idempotent, and requires no new dependency since createrepo_c is already linked.

```diff
diff --git a/src/createrepo-cache/repo_cache.c b/src/createrepo-cache/repo_cache.c
index 8c279ba..5d2bacb 100644
--- a/src/createrepo-cache/repo_cache.c
+++ b/src/createrepo-cache/repo_cache.c
@@ -424,6 +424,11 @@ cra_cache_new(const char * path)
   cra_Cache * cache;
   gpgme_error_t rc;
 
+  // Initialize libxml2 on the main thread. Without this, concurrent first-use
+  // from flush worker threads races on libxml2's lazy global init and crashes
+  // inside xmlDocDumpFormatMemoryEnc. The initialization is idempotent.
+  cr_xml_dump_init();
+
   cache = g_new0(cra_Cache, 1);
   if (!cache) {
     return NULL;
```

**Comment by j-rivero** [comment]
we need to rebase this one and solve the conflicts

**Review by j-rivero — COMMENTED** [review]
Minor comment, almost good to go. Agents complains about no real test to cover the "force write even with no changes". Something like this for test_smoke.py should be enough:

```python 
def test_touch_forces_rewrite(mutable_populated_repo: Path) -> None:
    arch_path = mutable_populated_repo / 'x86_64'
    repomd_path = arch_path / 'repodata' / 'repomd.xml'

    assert repomd_path.is_file()
    old_repomd_contents = repomd_path.read_text()

    # A plain commit with nothing staged must not rewrite already-clean metadata.
    with createrepo_agent.Server(str(mutable_populated_repo)):
        with createrepo_agent.Client(str(mutable_populated_repo)) as c:
            c.commit()

    assert old_repomd_contents == repomd_path.read_text()

    # 'touch' must force a metadata rewrite even though nothing else changed.
    with createrepo_agent.Server(str(mutable_populated_repo)):
        with createrepo_agent.Client(str(mutable_populated_repo)) as c:
            c.touch(('x86_64',))
            c.commit()

    assert old_repomd_contents != repomd_path.read_text()
```

File: src/createrepo-cache/repo_cache.c (hunk 1 above)
Comment (cottsay) [reply src/createrepo-cache/repo_cache.c:2005]: This is intentional, and is actually implicit behavior throughout this project. Maybe I should sick Gemini on writing some actual project-level documentation to clarify stuff like this.

**Review by j-rivero — APPROVED** [review]
